frontend/src/static/js/v_authorship.js

```diff
@@ -103,9 +106,36 @@ window.vAuthorship = {
       this.totalLineCount = totalLineCount;
       res.sort((a, b) => b.lineCount - a.lineCount);
 
+      this.filesLineInfo = this.convertLineCountToString(filesLineInfoArray);
       this.files = res;
       this.isLoaded = true;
     },
+
+    addLineCountToFileType(path, lineCount, filesLineInfoArray) {
+      const out = {};
+      var fileType = path.split(".").pop();
+      out.fileType = (fileType.length === 0) ? "others" : fileType;
+      out.lineCount = lineCount;
+      var index = filesLineInfoArray.findIndex((x) => x.fileType === out.fileType);
+      if (index === -1) {
+        filesLineInfoArray.push(out);
+      } else {
+        filesLineInfoArray[index].lineCount += out.lineCount;
+      }
```

Comment:
same comment as below, why did you choose to store the data in 2 separate arrays instead of a simple javascript object?

to initialize the field, just do a 
```
let obj = {};
if(!obj[fileType]){ obj[fileType] = 0; }
obj[fileType] += lineCount;
```

Reply:
@ongspxm i've changed the implementation to use a ```filesInfoObj``` instead of 2 arrays as requested, and the initialisation has been done as specified above.